### PR TITLE
GRW-758 / Feature / Checkout tracking page views

### PR DESCRIPTION
**Pull request by amandaflod**
## What?

Add tracking for page views for the new Checkout pages.

## Why?

**Ticket(s): [GRW-758]**

[GRW-758]: https://hedvig.atlassian.net/browse/GRW-758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by robinandeer — COMMENTED** [review]
Nice stuff :) Just have some comments

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 1]
```diff
@@ -20,13 +28,46 @@ import { DocumentLinks } from './components/DocumentLinks'
...
+      trackOfferEvent(
+        EventName.CheckoutOpen,
+        selectedBundleVariant.bundle,
+        isReferralCodeUsed,
+        {
+          quoteCartId,
+        },
+      )
+    }
+  }, [isReferralCodeUsed, quoteCartId, selectedBundleVariant])
```
[the remark this answers is not in the document]
Comment (amandaflod) [reply src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:65]: I thought that putting it in a useEffect would combat that 🤔 I thought about putting it on the button too, but that does not necessarily guarantee that the page has opened, no?

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 2]
```diff
@@ -20,13 +28,46 @@ import { DocumentLinks } from './components/DocumentLinks'
...
+
+  if (!data || data.loading) {
+    return <LoadingPage loading />
+  }
```
[the remark this answers is not in the document]
Comment (amandaflod) [reply src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:69]: Oh no! Will fix!

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (robinandeer) [reply src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:65]: Hmm, well maybe not guaranteed but good enough I think

`useEffect` will run any time the items change in the dependency array and send another event. Not a big deal maybe.

[the remark this answers is not in the document]
Comment (amandaflod) [reply src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:65]: Yes, but the items in the dependency array are not supposed to change at this point, right?

[the remark this answers is not in the document]
Comment (robinandeer) [reply src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:65]: hehe yeah probably right, we can leave and try like this

**Review by robinandeer — APPROVED** [review]

**Review by robinandeer — APPROVED** [review]

**Pull request merged**